sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java

```diff
@@ -302,14 +332,15 @@ public synchronized boolean isRunning() {
         return isRunning.get();
     }
 
-    private void stopProcessing() {
-        partitionPumpManager.stopAllPartitionPumps();
-        // finally, remove ownerid from checkpointstore as the processor is shutting down
-        checkpointStore.listOwnership(fullyQualifiedNamespace, eventHubName, consumerGroup)
-            .filter(ownership -> identifier.equals(ownership.getOwnerId()))
-            .map(ownership -> ownership.setOwnerId(""))
-            .collect(Collectors.toList())
-            .flatMapMany(checkpointStore::claimOwnership)
-            .blockLast(Duration.ofSeconds(10)); // block until the checkpoint store is updated
+    private Mono<Void> stopProcessing() {
+        return Mono.when(
+            partitionPumpManager.stopAllPartitionPumps(),
+
+            // finally, remove ownerid from checkpointstore as the processor is shutting down
+            checkpointStore.listOwnership(fullyQualifiedNamespace, eventHubName, consumerGroup)
```

Comment:
Will this list ownership execute if it is not being subscribed to?

Reply:
we subscribe to it via `Mono.when`. I updated to code to be a bit more clear.